components/webgpu/lib.rs

```diff
@@ -77,6 +83,7 @@ struct WGPU {
     adapters: Vec<WebGPUAdapter>,
     // Track invalid adapters https://gpuweb.github.io/gpuweb/#invalid
     _invalid_adapters: Vec<WebGPUAdapter>,
+    device_to_adapter: HashMap<WebGPUDevice, WebGPUAdapter>,
```

Comment:
do we need this map?
it seems like we should just carry a JS reference to `WebGPUAdapter` from the device struct

Reply:
We don't use it right now. I will remove this map. The reference to the `Adapter` is already in place. I will update the pull request with this change. Thank you!